abacatepay/__init__.py

```diff
@@ -8,45 +8,78 @@
+from ._models import Product, Billing, Customer
 from ._constants import BILLING_KINDS, BILLING_METHODS, BASEURL, USERAGENT
 from ._exceptions import *
-import requests
+
+
+logger = logging.getLogger(__name__)
+
 
 class AbacatePay:
-  """
-  The AbacatePay sdk client.
-  """
-
-  def __init__(self, api_key: str):
-    self.__api_key = api_key
-
-  def request(self, url: str, method: Literal["GET", "POST", "PUT", "PATCH", "DELETE"] = "GET", **kwargs):
-    return requests.request(method, url,  headers={"Authorization": f"Bearer {self.__api_key}", "User-Agent": USERAGENT}, **kwargs)
-
-  def CreateBilling(self, products:list[Product], returnURL:str, completionUrl:str, methods:list[BILLING_METHODS]=['PIX'], frequency:BILLING_KINDS='ONE_TIME', customerId:str|None=None, customer:Customer|None=None):
-    return Billing(products, returnURL, completionUrl, self.__api_key, methods, frequency=frequency, customerId=customerId, customer=customer)
-
-  def listBills(self) -> list[IBilling]:
-    response = self.request(f"{BASEURL}/billing/list", method="GET")
-    try:
-        if response.status_code == 200:
-            return [IBilling(data=bill) for bill in response.json()['data']]
-        else:
-            raise_for_status(response)
-    except requests.exceptions.Timeout:
-        raise APITimeoutError(request=response)
-    except requests.exceptions.ConnectionError:
-        raise APIConnectionError(message="Connection error.", request=response)
+    """
+    The AbacatePay sdk client.
+    """
+
+    def __init__(self, api_key: str):
+        self.__api_key = api_key
+
+    def _request(
+        self,
+        url: str,
+        method: Literal["GET", "POST", "PUT", "PATCH", "DELETE"] = "GET",
+        **kwargs,
+    ):
+        return requests.request(
+            method,
+            url,
+            headers={
+                "Authorization": f"Bearer {self.__api_key}",
+                "User-Agent": USERAGENT,
+            },
+            **kwargs,
+        )
+
+    def create_billing(
+        self,
+        products: list[Product],
+        returnURL: str,
+        completionUrl: str,
+        methods: list[BILLING_METHODS] = ["PIX"],
+        frequency: BILLING_KINDS = "ONE_TIME",
+        customerId: str | None = None,
+        customer: Customer | None = None,
+    ):
+        return Billing(
+            products,
+            returnURL,
+            completionUrl,
+            self.__api_key,
+            methods,
+            frequency=frequency,
+            customerId=customerId,
+            customer=customer,
+        )
+
+    def list_bills(self) -> list[Billing]:
+        logger.warning(f"Listing bills with URL: {BASEURL}/billing/list")
+        response = self._request(f"{BASEURL}/billing/list", method="GET")
 
+        try:
+            if response.status_code == 200:
+                return [Billing(data=bill) for bill in response.json()["data"]]
+            else:
+                raise_for_status(response)
```

Comment:
I'm sorry, I didn't understand this change. It has a syntax error (Try with no except or finally clause) and removes a feature (Error handling) for connection issues